mitodl/open-discussions
# make tracking with new google analytics g-tags possible

#### Pre-Flight checklist

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3316

#### What's this PR do?
Google analytics no longer supports `analytics.js`, which the `react-ga` module uses. This upgrades our code to use `gtag.js` so that the code works with newly created google tags that begin with G-.  The old universal analytics tags which start with U- are also supported with gtag.js 

To test out the new system and ensure we get the same metrics for both, this pr adds a new GA_G_TRACKING_ID variable so we can run both at the same time initially. GA_TRACKING_ID and `react-ga` will be removed in a followup pr once we confirm that everything is working properly on production. 

The only channel on production that currently has a tracking tag hasn't had posts in two years, so we can upgrade channel-specific tracking in this pr.

#### How should this be manually tested?
Set GA_G_TRACKING_ID in your .env file.
Go to the network tab in developer tools and search for the tag.
You should see calls to https://www.google-analytics.com with the tag when you click around the app.

Set the GA tracking id for a channel from the admin (http://localhost:8063/admin/channels/)
Go to the network tab in developer tools and search for the tag.
You should see calls to https://www.google-analytics.com with the tag when you click around the app in that channel but not when you are in a different channel or on other pages.
